### PR TITLE
[Backend] Database Restructure

### PR DESCRIPTION
**Pull request by brixtuyor**
# Related Links
- http://127.0.0.1:8000/categories
- http://127.0.0.1:8000/categories/:id
- http://127.0.0.1:8000/results/:id

# Description
- Changed table name lessons -> questions
- Configured the codebase to adopt to the table name change
- Added new model, controller, seeder and migration for Checkpoint

# Notes
- Please get rid of the previous migrations by running `php artisan migrate:rollback`
- New migration and seeder files added, please run `php artisan migrate --seed`

# Screenshots
image
image

### CODE REVIEW

File: database/seeders/CategorySeeder.php [hunk 1]
```diff
@@ -8,13 +8,26 @@
...
+	 *
+	 * @return void
+	 */
+	public function run()
+	{
+		Category::insert([
+			[
+				'title' => 'Numbers',
+				'description' => 'Ready to learn how to count from 1-100+ with Japanese numbers? Let\'s take a look at the Japanese numbers 1 - 10 in the Sino-Japanese Number system, which is most common.',
+				'created_at' => now(),
```
Comment (roseaugusto) [inline database/seeders/CategorySeeder.php:22]: @brixtuyor no need to put timestamps such `created_at' => now()` cause this is auto generated when you create. (unless you put it in fillable then you need to specify value)

Do the same for the rest of the models and kindly check if this is working.

Comment (brixtuyor) [reply database/seeders/CategorySeeder.php:22]: @roseaugusto Initially, I did not include the timestamps in the seeders. However, whenever I run the seeder files, the timestamps in the database are all null

Comment (brixtuyor) [reply database/seeders/CategorySeeder.php:22]: Let me try one more time...

Comment (brixtuyor) [reply database/seeders/CategorySeeder.php:22]: @roseaugusto I tried many times but the timestamps do not auto generate

Comment (roseaugusto) [reply database/seeders/CategorySeeder.php:22]: @brixtuyor okay. This should apply anyways if we use `create` but you are using `insert` so for now, just ignore this comment. Thanks!

**Pull request merged**